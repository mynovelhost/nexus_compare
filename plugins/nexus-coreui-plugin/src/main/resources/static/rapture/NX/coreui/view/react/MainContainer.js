/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
