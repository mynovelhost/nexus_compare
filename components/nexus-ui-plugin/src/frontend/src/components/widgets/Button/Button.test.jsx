/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Button from './Button';

describe('Button', () => {
  const renderButton = (extraProps) => {
    let props = {
      name: 'b1',
      onChange: () => {
      },
      ...extraProps
    };
    let {container} = render(<Button {...props} />);
    return container;
  };

  it('renders correctly', () => {
    expect(renderButton()).toMatchSnapshot();
  });

  it('renders correctly as a primary button', () => {
    expect(renderButton({variant: 'primary'})).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    expect(renderButton({disabled: true})).toMatchSnapshot();
  });

  it('allows for its style to be overriden', () => {
    expect(renderButton({
      style: {
        border: 'none'
      }
    })).toMatchSnapshot();
  });
});
