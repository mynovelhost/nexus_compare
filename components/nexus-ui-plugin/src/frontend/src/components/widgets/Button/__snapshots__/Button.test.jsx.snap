// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button allows for its style to be overriden 1`] = `
<div>
  <button
    class="nx-btn"
    name="b1"
    type="input"
  />
</div>
`;

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="nx-btn"
    name="b1"
    type="input"
  />
</div>
`;

exports[`Button renders correctly as a primary button 1`] = `
<div>
  <button
    class="nx-btn nx-btn--primary"
    name="b1"
    type="input"
  />
</div>
`;

exports[`Button renders correctly when disabled 1`] = `
<div>
  <button
    class="nx-btn"
    disabled=""
    name="b1"
    type="input"
  />
</div>
`;
